src/middlewared/middlewared/service/crud_service.py

```diff
@@ -432,3 +432,22 @@ async def get_dependencies(self, id_, ignored=None):
                 }, **data)
 
         return dependencies
+
+    @private
+    async def process_data(self, data, prefix, extend=None, context=None):
```

Comment:
This method duplicates code from the datastore plugin.

It also has to be called manually.

I propose we just define new key: `datastore_extend_fk`

```
class NVMetHostSubsysService(CRUDService):
    class Config:
        datastore_extend_fk = ['host', 'subsys']
```
And it automatically extends the defined foreign key references by:
* Resolving referenced table name from the column metadata
* Resolving plugin name that corresponds to that table
* Calling the corresponding plugin's extend function.